feat: add support for new prereq tree

ecxyzzy opened the pull request:
## Summary

Update the prereq tree rendering code to handle the API's new prereq tree format. Also add additional metadata to the leaf nodes if present.

Note that this is also `no deploy` because this will not work with API prod, since I manually uploaded the old prereq trees to the prod DB to prevent AA prod from breaking. Whoever decides to review this will need to checkout this branch locally and point the `PETERPORTAL_REST_ENDPOINT` at https://staging-58.api-next.peterportal.org/.

## Test Plan

Start dev server locally and make sure things don't break.

Some test cases I ran:

![image](https://github.com/icssc/AntAlmanac/assets/89349085/cff0ac83-5265-4234-8448-968c7f1c7383)

![image](https://github.com/icssc/AntAlmanac/assets/89349085/7e9b6866-d633-46ee-a1a6-3cab4efb3cac)

![image](https://github.com/icssc/AntAlmanac/assets/89349085/b5c0d3e4-35b6-4c2b-afe9-c0b2d6b035d7)

## Issues

N/A

ecxyzzy committed:
feat: add support for new prereq tree

EricPedley commented on apps/antalmanac/src/components/RightPane/SectionTable/PrereqTree.tsx:
```diff
@@ -40,30 +46,37 @@ interface TreeProps {
                         <div className={'prereq-branch'}>
-                            {Object.prototype.hasOwnProperty.call(prerequisite, 'OR') ? 'one of' : 'all of'}
+                            {
+                                Object.entries(phraseMapping).filter(([x, _]) =>
```
super-nit: one-letter variable name `x`. Maybe do something like `propertyName`. I'm even having trouble coming up with something because I barely understand what `x` is supposed to be here.

EricPedley commented on apps/antalmanac/src/components/RightPane/SectionTable/PrereqTree.tsx:
```diff
@@ -40,30 +46,37 @@ interface TreeProps {
 
 const Tree: FC<TreeProps> = (props) => {
```
Not part of your diff, but I feel like `Tree` sucks as a name for this class. Maybe throw in a name change to `PrereqTreeDisplayNode` or something else if you can think of a better one.

ecxyzzy committed:
refactor: address requested changes

EricPedley reviewed (approved):
Approving rn but I'm assuming we need to closely coordinate when to merge this into prod so we can release the API change at the same time. Will coordinate on Discord.

ecxyzzy committed:
fix: add dept/courseNumber to CourseInfo

ecxyzzy committed:
fix: make value node checking safer

ecxyzzy committed:
fix: use department+courseNumber instead of id

For instance, this renders "I&C SCI 33" instead of "I&CSCI33" with the
new API.

EricPedley reviewed (approved):
lgtm
